pylint/checkers/classes.py

```diff
@@ -974,9 +974,10 @@ def _check_unused_private_variables(self, node: astroid.ClassDef) -> None:
 
     def _check_unused_private_attributes(self, node: astroid.ClassDef) -> None:
         for assign_attr in node.nodes_of_class(astroid.AssignAttr):
-            if not is_attr_private(assign_attr.attrname):
+            if not is_attr_private(assign_attr.attrname) or not isinstance(
+                assign_attr.expr, astroid.Name
+            ):
                 continue
-
```

Comment:
I would keep the empty line here. It's a good logical separator.

Reply:
fixed